[libc++] Add some private headers to libcxx.imp

atetubou opened the pull request:
https://github.com/llvm/llvm-project/pull/78295 dropped private headers in top level directory from libcxx.imp.

This PR re-adds them to libcxx.imp.

atetubou commented:
@ldionne @philnik777 could you take a look this PR?

ldionne commented on libcxx/utils/libcxx/header_information.py:
```diff
@@ -192,25 +192,27 @@ def is_modulemap_header(header):
     header
     for header in toplevel_headers
-    if not header.endswith(".h")
+    if not header.endswith(".h") and not header.startswith("__")
```
I would suggest introducing

```
def is_public_header(header):
  return not header.startswith("__")
```

Then, `public_headers` becomes `public_headers = [p for p in all_headers if is_public_header(p)]` and the definition of `module_headers` will also read more naturally.

atetubou replied:
Done.

ldionne reviewed (commented):
Please rebase onto `main`, I just landed a change this morning that doesn't check-in the `.imp` file anymore.

Thanks for the patch and noticing this issue!

atetubou committed:
git squash commit for update_libcxx_imp.

f9a8c469008f905ab65ff3a2df38ed223e4d3bfd
git squash commit for update_libcxx_imp.

e0c4b64298c1552ba5c260875be13922d8fa2707
git squash commit for update_libcxx_imp.

3971b55e6f80fe157a4aafe19dacd084f034807e
add top level headers starting from __ in libcxx.imp

3dcc3b7750395a05c943ecb5f065acfae1046e49
underscore

0ac78d763c4a1c019c3e92385d30da64d35b6134
fmt

274337a1e3b910f316662714b96bc155537b9227
is_public_header

atetubou committed:
hear

atetubou commented on libcxx/utils/libcxx/header_information.py:
```diff
@@ -161,6 +161,10 @@ def is_header(file):
 
 
+def is_public_header(header):
+    return "__" not in header and not header.startswith("ext/")
```
`not header.startswith("ext/")` is to prevent appearing `ext/*` headers in 
https://github.com/llvm/llvm-project/blob/main/libcxx/include/__std_clang_module

But I'm not sure that is intentional or not.

atetubou commented:
Thank you for review!
